langserver/workspace_refs.go

```diff
@@ -27,6 +29,17 @@ import (
 // calls.
 const workspaceReferencesTimeout = 15 * time.Second
 
+// XReferenceAddOp is a JSON Patch operation used by
+// workspace/xreferences. It is exported so the build server can efficiently
+// interact with it. The only other patch operation is to create the empty
+// location list.
+type XReferenceAddOp struct {
+	// OP should always be "add"
+	OP    string               `json:"op"`
+	Path  string               `json:"path"`
+	Value referenceInformation `json:"value"` // TODO(keegancsmith) needs to be exported type
```

Comment:
this will have to be fixed before intergrating it in sourcegraph.com's streaming support.